pkg/controllers/deprovisioning/drift.go

```diff
@@ -56,11 +59,23 @@ func (d *Drift) ShouldDeprovision(ctx context.Context, c *Candidate) bool {
 		c.Machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted).IsTrue()
 }
 
+// SortCandidates orders drifted nodes by when they've drifted
+func (d *Drift) filterAndSortCandidates(ctx context.Context, nodes []*Candidate) ([]*Candidate, error) {
+	candidates, err := filterCandidates(ctx, d.kubeClient, d.recorder, nodes)
+	if err != nil {
+		return nil, fmt.Errorf("filtering candidates, %w", err)
+	}
+	sort.Slice(candidates, func(i int, j int) bool {
+		return machineutil.GetDriftedTime(candidates[i].Machine).Before(machineutil.GetDriftedTime(candidates[j].Machine))
+	})
+	return candidates, nil
+}
+
 // ComputeCommand generates a deprovisioning command given deprovisionable machines
 func (d *Drift) ComputeCommand(ctx context.Context, nodes ...*Candidate) (Command, error) {
-	candidates, err := filterCandidates(ctx, d.kubeClient, d.recorder, nodes)
+	candidates, err := d.filterAndSortCandidates(ctx, nodes)
 	if err != nil {
```

Comment:
It's the same error wrap as before, just moved up into a different function. Since there's no way the sort can fail, I just decided to return the error.